wasmsdk/allocation.go

```diff
@@ -189,11 +196,88 @@ func getRemoteFileMap(allocationID string) ([]*fileResp, error) {
 	return fileResps, nil
 }
 
-func getJSON(v interface{}) (string, error) {
-	b, err := json.Marshal(v)
+// WritePoolLock locks given number of tokes for given duration in read pool.
+// ## Inputs
+//   - allocID: allocation id
+//   - tokens:  sas tokens
+//   - fee: sas tokens
+func writePoolLock(allocID, tokens, fee string) (string, error) {
+	t, err := util.ParseCoinStr(tokens)
+	if err != nil {
+		return "", err
+	}
+
+	f, err := util.ParseCoinStr(fee)
+	if err != nil {
+		return "", err
+	}
+	hash, _, err := sdk.WritePoolLock(allocID, t, f)
+	return hash, err
+}
+
+// GetReadPoolInfo is to get information about the read pool for the allocation
+func getReadPoolInfo(clientID string) (*sdk.ReadPool, error) {
+	readPool, err := sdk.GetReadPoolInfo(clientID)
+	if err != nil {
+		return nil, err
+	}
+
+	return readPool, nil
+}
+
+// GetAllocationFromAuthTicket - get allocation from Auth ticket
+func getAllocationFromAuthTicket(authTicket string) (*sdk.Allocation, error) {
+	sdkAllocation, err := sdk.GetAllocationFromAuthTicket(authTicket)
+	if err != nil {
+		return nil, err
+	}
+	return sdkAllocation, err
+}
+
+func redeemFreeStorage(ticket string) (string, error) {
+	recipientPublicKey, marker, lock, err := decodeTicket(ticket)
 	if err != nil {
-		sdkLogger.Error("Failed to convert data to json format : %v", err)
 		return "", err
 	}
-	return string(b), nil
+
+	if recipientPublicKey != client.GetClientPublicKey() {
+		return "", fmt.Errorf("invalid_free_marker: free marker is not assigned to your wallet")
+	}
+
+	hash, _, err := sdk.CreateFreeAllocation(marker, lock)
```

Comment:
it doesn't work anymore. please use 0box api to create free allocation.

Reply:
removed freeallocation from wasm in 5657e95